### PR TITLE
Fix #4773 - separate RNA view genome build option

### LINKED ISSUE #4773: RNA view hg19 aln
**Is your feature request related to a problem in the current program to new available techology or software? Please describe and add links/citations if appropriate.**
Sometimes researchers choose to use hg19 for their RNA projects.

**Describe the solution you'd like**
Differentiate the RNA aln view to work also for hg19 RNA files.

### PR DESCRIPTION
**Pull request by dnil**
This PR adds a functionality or fixes a bug.

One can now view RNA alns in hg19 by setting `rna_human_genome_build: 37` in the load config.
Screenshot 2024-08-23 at 14 11 55

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. how to test it, possibly with real cases/data

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [ ] tests executed by

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -24,6 +24,7 @@ About changelog [here](https://keepachangelog.com/en/1.0.0/)
 - Possibility to un-audit previously audited filters
 - Reverted table style and removed font awesome style from IGV template
 - Case status tags displayed on dashboard case overview
+- Option to configure RNA build on case load (default '38')
```
Comment (northwestwitch) [inline CHANGELOG.md:27]: Careful with the changelog here

File: scout/server/blueprints/alignviewers/controllers.py [hunk 2]
```diff
@@ -180,6 +171,30 @@ def make_sashimi_tracks(
     return display_obj
 
 
+def make_merged_splice_track(ind: dict) -> dict:
```
Comment (dnil) [inline scout/server/blueprints/alignviewers/controllers.py:174]: Moved to a separate auction to comment and reduce cognitive load.

File: scout/models/case/case_loading_models.py [hunk 3]
```diff
@@ -499,6 +500,20 @@ def set_synopsis(cls, synopsis: Optional[Union[str, List]]) -> Optional[str]:
     @model_validator(mode="before")
     @classmethod
     def format_build(cls, values) -> "CaseLoader":
```
Comment (northwestwitch) [inline scout/models/case/case_loading_models.py:502]: Shouldn't this be named `format_rna_build` and the function down (line 516) `format_build`?

Comment (northwestwitch) [reply scout/models/case/case_loading_models.py:502]: Also these 2 functions have some common code. Couldn't they be unified somehow?

File: scout/server/blueprints/alignviewers/controllers.py [hunk 4]
```diff
@@ -141,6 +140,10 @@ def make_sashimi_tracks(
     if omics_variant_id:
         variant_obj = store.omics_variant(variant_id=omics_variant_id)
 
+    build = "38"
+    if "37" in str(case_obj.get("rna_genome_build", "38")):
```
Comment (northwestwitch) [inline scout/server/blueprints/alignviewers/controllers.py:144]: 👍🏻

**Review by northwestwitch — APPROVED** [review]
Looks good, just those  pydantic checks that should at least swap names

File: scout/models/case/case_loading_models.py (hunk 3 above)
Comment (dnil) [reply scout/models/case/case_loading_models.py:502]: Sure, one way would be to break out the genome string check to a separate helper..

**Pull request merged**